# U.S. Medical Insurance Costs

The purpose of this project is to analyze a csv file called insurance.csv, in order to have clear information about the cost of insurance in the United States according to several criteria represented in our csv file.To complete the project, I will use the skills I have learned in previous classes.

In [1]:
# We will mainly use the python library pandas for data analysis.
#importation of pandas' module
import pandas as pd

## Recovery of data.

In [2]:
#read_csv allows us to read a csv file with pandas
insurance_cost_data = pd.read_csv('insurance.csv')

## Examine data structure


Let's look at the first 50 lines using the head() method of the DataFrame

In [3]:
insurance_cost_data.head(50)

age     sex     bmi  children smoker     region      charges
0    19  female  27.900         0    yes  southwest  16884.92400
1    18    male  33.770         1     no  southeast   1725.55230
2    28    male  33.000         3     no  southeast   4449.46200
3    33    male  22.705         0     no  northwest  21984.47061
4    32    male  28.880         0     no  northwest   3866.85520
5    31  female  25.740         0     no  southeast   3756.62160
6    46  female  33.440         1     no  southeast   8240.58960
7    37  female  27.740         3     no  northwest   7281.50560
8    37    male  29.830         2     no  northeast   6406.41070
9    60  female  25.840         0     no  northwest  28923.13692
10   25    male  26.220         0     no  northeast   2721.32080
11   62  female  26.290         0    yes  southeast  27808.72510
12   23    male  34.400         0     no  southwest   1826.84300
13   56  female  39.820         0     no  southeast  11090.71780
14   27    male  42.130         0    yes  southeast  39611.75770
15   19    male  24.600         1     no  southwest   1837.23700
16   52  female  30.780         1     no  northeast  10797.33620
17   23    male  23.845         0     no  northeast   2395.17155
18   56    male  40.300         0     no  southwest  10602.38500
19   30    male  35.300         0    yes  southwest  36837.46700
20   60  female  36.005         0     no  northeast  13228.84695
21   30  female  32.400         1     no  southwest   4149.73600
22   18    male  34.100         0     no  southeast   1137.01100
23   34  female  31.920         1    yes  northeast  37701.87680
24   37    male  28.025         2     no  northwest   6203.90175
25   59  female  27.720         3     no  southeast  14001.13380
26   63  female  23.085         0     no  northeast  14451.83515
27   55  female  32.775         2     no  northwest  12268.63225
28   23    male  17.385         1     no  northwest   2775.19215
29   31    male  36.300         2    yes  southwest  38711.00000
30   22    male  35.600         0    yes  southwest  35585.57600
31   18  female  26.315         0     no  northeast   2198.18985
32   19  female  28.600         5     no  southwest   4687.79700
33   63    male  28.310         0     no  northwest  13770.09790
34   28    male  36.400         1    yes  southwest  51194.55914
35   19    male  20.425         0     no  northwest   1625.43375
36   62  female  32.965         3     no  northwest  15612.19335
37   26    male  20.800         0     no  southwest   2302.30000
38   35    male  36.670         1    yes  northeast  39774.27630
39   60    male  39.900         0    yes  southwest  48173.36100
40   24  female  26.600         0     no  northeast   3046.06200
41   31  female  36.630         2     no  southeast   4949.75870
42   41    male  21.780         1     no  southeast   6272.47720
43   37  female  30.800         2     no  southeast   6313.75900
44   38    male  37.050         1     no  northeast   6079.67150
45   55    male  37.300         0     no  southwest  20630.28351
46   18  female  38.665         2     no  northeast   3393.35635
47   28  female  34.770         0     no  northwest   3556.92230
48   60  female  24.530         0     no  southeast  12629.89670
49   36    male  35.200         1    yes  southeast  38709.17600

Every line represents an observation our dataset.

In [5]:
insurance_cost_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


The info() method allows to obtain a description of the data, in particular
the total number of rows, the type of each variable and the number of non
null values

### We have three categorical variables in our dataset: sex, smoker, and region.

We can obtain the list of the modalities
of these categorical variables as well as the counts of each of them (i.e. the variables sex, smoker and region
in each category) by using the value_counts() method:

In [16]:
print(f"Sex :\n {insurance_cost_data['sex'].value_counts()}\n")
print(f"Smoker :\n {insurance_cost_data['smoker'].value_counts()}\n")
print(f"Region :\n {insurance_cost_data['region'].value_counts()}\n")

Sex :
 male      676
female    662
Name: sex, dtype: int64

Smoker :
 no     1064
yes     274
Name: smoker, dtype: int64

Region :
 southeast    364
northwest    325
southwest    325
northeast    324
Name: region, dtype: int64



Let's look at the other fields: they are all quantitative variables or
variables. The method describe()presents a summary of the numerical variables

In [17]:
insurance_cost_data.describe()

age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010

The lines count, mean, min and max provide the number of values, the mean, the minimum and the maximum respectively.
values, the average, the minimum and the maximum respectively.
The std line indicates the standard deviation (which
the dispersion of the data)26. The lines 25%, 50% and 75% show the corresponding
percentiles: a percentile indicates the value below which a given percentage of the observations in the group falls.
percentage of the observations in the group.
Per example mean's age of patients is 39.207025, and patients have in mean 1.094918 child.

### Analyze the origin of the majority of the individuals.

In [23]:
insurance_cost_data.groupby(['region']).mean()

age        bmi  children       charges
region                                                 
northeast  39.268519  29.173503  1.046296  13406.384516
northwest  39.196923  29.199785  1.147692  12417.575374
southeast  38.939560  33.355989  1.049451  14735.411438
southwest  39.455385  30.596615  1.141538  12346.937377